bn254-fr/src/poseidon2.rs

```diff
@@ -89,9 +89,32 @@ mod tests {
             })
             .collect();
 
+        let external_round_constants: Vec<[F; WIDTH]> = round_constants
+            .iter()
+            .enumerate()
+            .filter_map(|(i, val)| {
+                if ((ROUNDS_F / 2)..((ROUNDS_F / 2) + ROUNDS_P)).contains(&i) {
+                    None
+                } else {
+                    Some(*val)
+                }
+            })
+            .collect();
+        let internal_round_constants: Vec<F> = round_constants
+            .iter()
+            .skip(ROUNDS_F / 2)
+            .take(ROUNDS_P)
+            .map(|vec| vec[0])
+            .collect();
+
```

Comment:
Shorter, clearer, and more idiomatic:
```suggestion
        let internal_start = ROUNDS_F / 2;
        let internal_end = (ROUNDS_F / 2) + ROUNDS_P;
        // need to make `round_constants` mutable above for this to work
        let internal_round_constants = round_constants
            .drain(internal_start..internal_end)
            .map(|vec| vec[0])
            .collect::<Vec<_>>();
        // optional rename; probably simpler to just to call the original variable `external_round_constants`
        let external_round_constants = round_constants;
```